samza-test/src/test/java/org/apache/samza/test/framework/AsyncStreamTaskIntegrationTest.java

```diff
@@ -44,10 +49,82 @@ public void testAsyncTaskWithSinglePartition() throws Exception {
         .addOutputStream(output)
         .run(Duration.ofSeconds(2));
 
-    Assert.assertThat(TestRunner.consumeStream(output, 1000).get(0),
+    Assert.assertThat(TestRunner.consumeStream(output, Duration.ofMillis(1000)).get(0),
         IsIterableContainingInOrder.contains(outputList.toArray()));
   }
 
+  @Test
+  public void testAsyncTaskWithSinglePartitionUsingStreamAssert() throws Exception {
+    List<Integer> inputList = Arrays.asList(1, 2, 3, 4, 5);
+    List<Integer> outputList = Arrays.asList(50, 10, 20, 30, 40);
+
+    CollectionStream<Integer> input = CollectionStream.of("async-test", "ints", inputList);
+    CollectionStream output = CollectionStream.empty("async-test", "ints-out");
+
+    TestRunner
+        .of(MyAsyncStreamTask.class)
+        .addInputStream(input)
+        .addOutputStream(output)
+        .run(Duration.ofSeconds(2));
+
+    StreamAssert.that(output).containsInAnyOrder(outputList, Duration.ofMillis(1000));
+  }
+
+  @Test
+  public void testAsyncTaskWithMultiplePartition() throws Exception {
+    Map<Integer, List<KV>> input = new HashMap<>();
+    Map<Integer, List<Integer>> output = new HashMap<>();
+    List<Integer> partition = Arrays.asList(1, 2, 3, 4, 5);
+    List<Integer> outputPartition = partition.stream().map(x -> x * 10).collect(Collectors.toList());
+    for (int i = 0; i < 5; i++) {
+      List<KV> keyedPartition = new ArrayList<>();
+      for (Integer val : partition) {
+        keyedPartition.add(KV.of(i, val));
+      }
+      input.put(i, keyedPartition);
+      output.put(i, new ArrayList<Integer>(outputPartition));
+    }
+
+    CollectionStream<KV> inputStream = CollectionStream.of("async-test", "ints", input);
+    CollectionStream outputStream = CollectionStream.empty("async-test", "ints-out", 5);
+
+    TestRunner
+        .of(MyAsyncStreamTask.class)
+        .addInputStream(inputStream)
+        .addOutputStream(outputStream)
+        .run(Duration.ofSeconds(2));
+
+    StreamAssert.that(outputStream).contains(output, Duration.ofMillis(1000));
+  }
+
+  @Test
+  public void testAsyncTaskWithMultiplePartitionMultithreaded() throws Exception {
+    Map<Integer, List<KV>> input = new HashMap<>();
+    Map<Integer, List<Integer>> output = new HashMap<>();
+    List<Integer> partition = Arrays.asList(1, 2, 3, 4, 5);
+    List<Integer> outputPartition = partition.stream().map(x -> x * 10).collect(Collectors.toList());
+    for (int i = 0; i < 5; i++) {
+      List<KV> keyedPartition = new ArrayList<>();
+      for (Integer val : partition) {
+        keyedPartition.add(KV.of(i, val));
+      }
+      input.put(i, keyedPartition);
+      output.put(i, new ArrayList<Integer>(outputPartition));
```

Comment:
Since there is just one input/output message per partition, this doesn't really seem to be testing per-task multithreading. Could you add more inputs/outputs per partition?

Reply:
There are 5 messages per partition and 5 partitions to test, not sure what the confusion is.

Reply:
You are right. I misread it before. I got mixed up where the data was coming from.